datafusion-cli/src/print_format.rs

```diff
@@ -159,14 +191,41 @@ impl PrintFormat {
         }
         Ok(())
     }
+
+    pub fn print_stream(&self, batch: &RecordBatch, with_header: bool) -> Result<()> {
+        if batch.num_rows() == 0 {
+            return Ok(());
+        }
+
+        match self {
+            Self::Csv | Self::Automatic => {
+                println!("{}", print_stream_with_sep(batch, b',', with_header)?)
+            }
+            Self::Tsv => {
+                println!("{}", print_stream_with_sep(batch, b'\t', with_header)?)
+            }
+            Self::Table => {
```

Comment:
Another thought here is that when printing streams, it could just print out a new Table for each batch (rather than buffering all batches).